### PR TITLE
Renamed the gem to logging_library

### PR DESCRIPTION
**Pull request by perlun**
Also added an explicit MIT license, since this library is now released as [FOSS](https://en.wikipedia.org/wiki/Free_and_open-source_software)

The PR consists mostly of files being renamed, but I also added a compatibility spec for ensuring that the `Ecraft::LoggingLibrary` module still exists.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ push git commits and tags, and push the `.gem` file to
 
 ## Contributing
 
-Bug reports and pull requests are welcome on GitHub at 
+Bug reports and pull requests are welcome on GitHub at
 https://github.com/ecraft/ecraft-logging_library. This project is intended to
```
Comment (smgt) [inline README.md]: Fixa urlen!

Comment (perlun) [inline README.md]: Thanks. The Usage also refers to stuff using the old namespace, will fix.

Comment (smgt) [inline README.md]: Haha, was just about to comment on that. ;)

Comment (perlun) [inline README.md]: :) Fixed the errors, please recheck and I'll merge in a couple of minutes.

**Pull request merged**